client.py

```diff
@@ -93,18 +91,22 @@ def get_authn_req_url(self, session, acr, forceAuthN, scope):
         """
         state = tools.generate_random_string()
         session['state'] = state
-        session['code_verifier'] = code_verifier = tools.generate_random_string(100)
+        session['code_verifier'] = code_verifier = tools.generate_random_string(100).encode()
 
-        code_challenge = tools.base64_urlencode(hashlib.sha256(code_verifier).digest())
+        code_challenge = self.__create_code_challenge(code_verifier)
```

Comment:
hard to tell; where is this method?

Reply:
It's on line 106. Its the above line but moved to a function. I had more changes in there before, but the result became the same thing. So I could revert it, but it also reads a little bit easier like this I think.

Reply:
This is fine. I just missed the method before in the GH view of the code.